SessionMessagingKit/Jobs/Types/GroupLeavingJob.swift

```diff
@@ -0,0 +1,167 @@
+// Copyright © 2023 Rangeproof Pty Ltd. All rights reserved.
+
+import Foundation
+import GRDB
+import PromiseKit
+import SignalCoreKit
+import SessionUtilitiesKit
+import SessionSnodeKit
+
+public enum GroupLeavingJob: JobExecutor {
+    public static var maxFailureCount: Int = 0
+    public static var requiresThreadId: Bool = true
+    public static var requiresInteractionId: Bool = true
+    
+    public static func run(
+        _ job: SessionUtilitiesKit.Job,
+        queue: DispatchQueue,
+        success: @escaping (SessionUtilitiesKit.Job, Bool) -> (),
+        failure: @escaping (SessionUtilitiesKit.Job, Error?, Bool) -> (),
+        deferred: @escaping (SessionUtilitiesKit.Job) -> ())
+    {
+        print("Ryan Test: Group leaving job running")
+        guard
+            let detailsData: Data = job.details,
+            let details: Details = try? JSONDecoder().decode(Details.self, from: detailsData),
+            let interactionId: Int64 = job.interactionId
+        else {
+            failure(job, JobRunnerError.missingRequiredDetails, false)
```

Comment:
Hmm... I wonder if we should be updating all of the jobs to be permanent failures in this case (if they don't have the required data then they would always fail right?)

Reply:
Hmm... yeah, I think so. I'll update all of them.